etl/src/main/java/org/pih/hivmigration/etl/sql/ProgramMigrator.groovy

```diff
@@ -0,0 +1,252 @@
+                    NULL,
+                    starting_health_center)
+            ;
+        ''')
+
+        executeMysql("Load single enrollments to hivmigration_programs table", '''
+            INSERT INTO hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT source_patient_id, health_center, enrollment_date, art_start_date, outcome_date, outcome FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NULL
+            ;
+        ''')
+
+        executeMysql("Load initial enrollments into hivmigration_programs table", '''
+            insert into hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT
+                source_patient_id,
+                starting_health_center,
+                enrollment_date,
+                IF (art_start_date < health_center_transfer_date, art_start_date, NULL),
+                health_center_transfer_date,
+                'TRANSFERRED_INTERNALLY'
+            FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NOT NULL
+            ;
+        ''')
+
+        executeMysql("Load current enrollments into hivmigration_programs table", '''
+            insert into hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT
+                source_patient_id,
+                health_center,
+                health_center_transfer_date,
+                IF (art_start_date < health_center_transfer_date, health_center_transfer_date, art_start_date),
+                outcome_date,
+                outcome
+            FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NOT NULL
+            ;
+        ''')
+
+        executeMysql("Load to patient_program table", '''
+            insert into patient_program
+                (patient_program_id, patient_id, program_id, date_enrolled, date_completed, location_id, outcome_concept_id, creator, date_created, uuid)
+            select
+                h.patient_program_id, 
+                p.person_id,
+                2,
+                h.enrollment_date, 
+                h.outcome_date, 
+                hc.openmrs_id,
+                o.outcome_concept_id, 
+                1,
+                date_format(curdate(), '%Y-%m-%d %T'),
+                uuid()
+            from 
+                hivmigration_programs h 
+            inner join 
+                hivmigration_patients p on h.source_patient_id = p.source_patient_id 
+            left outer join
+                hivmigration_health_center hc on h.location_id = hc.hiv_emr_id
+            left outer join 
+                hivmigration_outcome o on h.outcome = o.outcome
+            ; 
+        ''')
+        
+        executeMysql("Add patient state", '''
```

Comment:
I'm not entirely sure what this is doing, but it seems wrong.  If we are inserting a state for patients who do not have an ART start date or for whom their ART start date > enrollment date, then this means that we might need to insert 2 states for these patients to reflect their state when they started ART.

This needs more documentation and discussion / review.

Reply:
Yeah, looks like this is just entering one state per patient program? We should re-review with Ellen and others what the states going forward and how we want the mapping to work.